src/main/scala/shell/IOShell.scala

```diff
@@ -15,7 +15,7 @@ case class IOPin(element: Element, index: Int = 0)
   require (index >= 0 && index < width.get)
 
   def name = {
-    val pin = element.instanceName.split("\\.").mkString("_")
+    val pin = element.instanceName.split("\\.").map(_.replaceAll("""\[(\d+)\]""", "_$1")).mkString("_")
```

Comment:
For the record I think this is in the realm of needing a comment explaining what's going on

Reply:
yeah, I will add one